Add migration docs (#9)
exercism/v3-launch: migration/build-analyzer-and-representer.md

Once we have a normalized representation for a solution, a team of vetted mentors will look at the solution and comment on it (if needed). These comments will then automatically be submitted to each new solution with the same representation. A notification will be sent for old solutions with a matching representation.

Whenever a solution is submitted, a Docker container is created of the representer image and it runs on the submitted solution. The test runner outputs two JSON files that describe the representation. See [the spec](https://github.com/exercism/v3-docs/blob/master/anatomy/track-tooling/representers/interface.md).

The representer is an optional tool, which means that if a track does not have a representer, it will still function normally.

## Analyzer

> In Exercism V3, we're introducing a new (optional) tool: the [_analyzer_](https://github.com/exercism/v3-docs/tree/master/anatomy/track-tooling/analyzers). Analyzers automatically assess student's submissions and provide mentor-style commentary. They can be used to catch common mistakes and/or do complex solution analysis that can't easily be done directly in a test suite.

Reviewer: ```suggestion
In Exercism V3, we are making increased use of our V2 [_analyzers_](https://github.com/exercism/v3-docs/tree/master/anatomy/track-tooling/analyzers). Analyzers automatically assess student's submissions and provide mentor-style commentary. They can be used to catch common mistakes and/or do complex solution analysis that can't easily be done directly in a test suite.

In V2, analyzer comments were given to a mentor to pass to a student. In V3, the analyzers will normally output directly to students, although we have added an extra key to output suggestions to mentors. If your track already  has an analyzer, the only requisite change is updating the outputted copy to be student-facing.

```

Erik - I need to update the docs about how analyzer commentry should be written to students.

Whenever a solution is submitted, a Docker container is created of the analyzer image and it runs on the submitted solution. The analyzer outputs a JSON file that contains the analysis results. See [the spec](https://github.com/exercism/v3-docs/blob/master/anatomy/track-tooling/analyzers/interface.md).

> Note that some tracks have already implemented an experimental V2 analyzer, which the V3 analyzer is an evolution of.

Reviewer: I've sort of included this in the paragraph above now. You might want to work out how they sit together.

The analyzer is an optional tool, which means that if a track does not have an analyzer, it will still function normally.

## Choosing between representer and analyzer

If you want to build both, our recommendation is to start building the representer first for the following reasons:

- Representers are usually (far) easier to implement

> - Representers can have a far bigger impact on the mentoring load than analyzers

Reviewer: ```suggestion
- Representers can have a far bigger impact on the mentoring load than analyzers by empowering mentors

```

> # Build Analyzer and Representer

Reviewer: ```suggestion
# Build Representer and Analyzer
```

(match the order below)

Each analyzer is track-specific. When a new solution is submitted, we run the track's analyzer, which outputs a JSON file that contains the analysis results.

In v2, analyzer comments were given to a mentor to pass to a student. In v3, the analyzers will normally output directly to students, although we have added an extra key to output suggestions to mentors. If your track already has an analyzer, the only requisite change is updating the outputted copy to be student-facing.

Each track should build an analyzer according to [the spec](https://github.com/exercism/v3-docs/blob/master/anatomy/track-tooling/analyzers/interface.md). For tracks building an analyzer from scratch, we have a [starting guide](https://github.com/exercism/v3-docs/blob/master/anatomy/track-tooling/analyzers/creating-from-scratch.md).

The analyzer is an optional tool though, which means that if a track does not have an analyzer, it will still function normally.

## Choosing between representer and analyzer

> If you want to build both, our recommendation is to start building the representer first for the following reasons:

Reviewer: ```suggestion
We recommend starting by building the representer for the following reasons:

```

Reviewer: (This bit is relevant if they want to build either one or both)
